tests/register/register.go

```diff
@@ -31,6 +39,33 @@ func init() {
 	Tests = make(map[TestType][]types.Test)
 }
 
-func Register(tType TestType, t types.Test) {
+func register(tType TestType, t types.Test) {
 	Tests[tType] = append(Tests[tType], t)
 }
+
+// Registers t for every version, inside the same major version, that is equal to or greater than the specified ConfigMinVersion.
+func Register(tType TestType, t types.Test) {
+	// update confgiVersions with new config versions
+	configVersions := [][]semver.Version{
+		{semver.Version{}}, // place holder
+		{types1.MaxVersion},
+		{types20.MaxVersion, types21.MaxVersion, types22.MaxVersion, types23.MaxVersion},
+	}
+
+	version, semverErr := semver.NewVersion(t.ConfigMinVersion)
+
+	register(tType, t)
+	if semverErr == nil && version != nil && t.ConfigMinVersion != "" {
+		initalVersion := *version
+		for *version != configVersions[version.Major][len(configVersions[version.Major])-1] {
```

Comment:
My idea for how this could work would be to bump the `go-semver` version to get access to the [`LessThan`](https://github.com/coreos/go-semver/blob/master/semver/semver.go#L160-L162) function. Then this could be simplified to something like:

```golang
for _, v := range configVersions[version.Major] {
    if version.LessThan(v) {
        t := types.DeepCopy(t)
        t.Config = strings.Replace(t.Config, "$version", v.String(), -1)
        register(tType, t)
    }
}
```

Reply:
That makes sense. Thanks for the example!